.ci/check_python_dists.sh

```diff
@@ -17,4 +17,35 @@ if { test "${TASK}" = "bdist" || test "${METHOD}" = "wheel"; }; then
     check-wheel-contents ${DIST_DIR}/*.whl || exit -1
 fi
 
+PY_MINOR_VER=$(python -c "import sys; print(sys.version_info.minor)")
+if [ $PY_MINOR_VER -gt 7 ]; then
+    echo "pydistcheck..."
+    pip install pydistcheck
+    if [ $TASK == "CUDA" ] && [ $METHOD == "wheel" ]; then
+        pydistcheck \
+            --inspect \
+            --ignore 'compiled-objects-have-debug-symbols,max-allowed-size-compressed' \
+            --max-allowed-size-uncompressed '60M' \
+            --max-allowed-files 800 \
+            ${DIST_DIR}/* || exit -1
+    elif [ $ARCH == "aarch64" ]; then
+        pydistcheck \
+            --inspect \
+            --ignore 'compiled-objects-have-debug-symbols' \
+            --max-allowed-size-compressed '5M' \
+            --max-allowed-size-uncompressed '15M' \
+            --max-allowed-files 800 \
+            ${DIST_DIR}/* || exit -1
```

Comment:
The aarch64 wheel we produces one error from `pydistcheck`:

```text
1. [compiled-objects-have-debug-symbols] Found compiled object containing debug symbols. For details, extract the distribution contents and run 'nm --debug-syms "lightgbm/lib_lightgbm.so"'.
errors found while checking: 1
```

([build link](https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=14565&view=logs&j=c2f9361f-3c13-57db-3206-cee89820d5e3&t=bf5b33f9-9072-549e-d8b1-cab79a1dd2a9))

I'm not sure where those are coming from, but I'm proposing ignoring them for now. The aarch64 linux integrated OpenCL wheel has a very similar compressed size (2.3MB) to the arm64 linux wheels (1.6M).